engine/src/main/java/org/camunda/bpm/engine/HistoryService.java

```diff
@@ -107,21 +109,58 @@
    */
   HistoricCaseActivityStatisticsQuery createHistoricCaseActivityStatisticsQuery(String caseDefinitionId);
 
-  /** Creates a new programmatic query to search for {@link HistoricTaskInstance}s. */
+  /**
+   * <p>Creates a new programmatic query to search for {@link HistoricTaskInstance}s.
+   *
+   * <p>The result of the query is empty when the user has no...
+   * <ul>
+   *   <li>{@link Permissions#READ_HISTORY} permission on {@link Resources#PROCESS_DEFINITION} OR
+   *   <li>{@link HistoricTaskPermissions#READ} permission on
+   *       {@link Resources#HISTORIC_TASK} ({@code enableHistoricInstancePermissions} in
+   *       {@link ProcessEngineConfigurationImpl} must be set to {@code true})
+   * */
   HistoricTaskInstanceQuery createHistoricTaskInstanceQuery();
 
-  /** Creates a new programmatic query to search for {@link HistoricDetail}s. */
+  /**
+   * <p>Creates a new programmatic query to search for {@link HistoricDetail}s.
+   *
+   * <p>The result of the query is empty when the user has no...
+   * <ul>
+   *   <li>{@link Permissions#READ_HISTORY} permission on
+   *       {@link Resources#PROCESS_DEFINITION} OR
+   *   <li>{@link HistoricTaskPermissions#READ} permission on
+   *       {@link Resources#HISTORIC_TASK} ({@code enableHistoricInstancePermissions} in
+   *       {@link ProcessEngineConfigurationImpl} must be set to {@code true}) OR
+   *   <li>{@link ProcessDefinitionPermissions#READ_HISTORY_VARIABLE} permission on
+   *       {@link Resources#PROCESS_DEFINITION}
+   *       ({@link ProcessEngineConfiguration#enforceSpecificVariablePermission} must be set to
```

Comment:
to unify with the rest of the mentions
```suggestion
   *       ({@code enforceSpecificVariablePermission} in {@link ProcessEngineConfigurationImpl} must be set to
```